src/akar/util.clj

```diff
@@ -0,0 +1,42 @@
+(ns akar.util
+  (:require [clojure.walk :as cw]))
+
+(defn postwalk-fn-replace
+  "Calls the matching-fn on every expression and if matching-fn
+  returned true, replaces the expression by the result of calling
+  replacing-fn with the given expr as the parameter"
+  [matching-fn replacing-fn form]
+  (cw/postwalk (fn [x]
+                   (if (matching-fn x)
+                     (replacing-fn x)
+                     x))
+                 form))
+
+
+(defmacro def-trampolined-recur
+  "For recursive functions using recur along with akar match clauses,
+  this macro generates a tail call optimized function body using
+  trampoline.
+
+  This generates two functions,
+  1. Function with \"*\" suffixed to the name give. The body for
+  this function is same as the body given except the recur call replaced
+  with an anonymous function calling the same function recursively.
+
+  2. A function with the name - same as the name given in parameter. This
+  function calls the above generated function using trampoline.
+
+  This is done to avail the tail call optimization along with using the akar
+  patterns"
+  [nm args & body]
```

Comment:
I understand that you are calling the argument `nm` to avoid clashing with function `name`. Is this in accordance with Clojure conventions? Or might something like `name'` work better?

Reply:
I have renamed he variable `nm` to `fn-name` to avoid the confusion. I checked online but could not find any particular convention to be followed in such scenarios.